DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs

```diff
@@ -150,6 +156,39 @@ public IEnumerable<GroupCourse> GetGroupCourses(int groupId, int centreId)
             );
         }
 
+        public GroupCourse? GetGroupCourse(int groupCustomisationId, int groupId, int centreId)
+        {
+            return connection.Query<GroupCourse>(
+                @"SELECT
+                        GroupCustomisationID,
+                        GroupID,
+                        gc.CustomisationID,
+                        ap.ApplicationName,
+                        CustomisationName,
+                        Mandatory AS IsMandatory,
+                        IsAssessed,
+                        AddedDate AS AddedToGroup,
+                        c.CurrentVersion,
+                        gc.SupervisorAdminID,
+                        au.Forename AS SupervisorFirstName,
+                        au.Surname AS SupervisorLastName,
+                        gc.CompleteWithinMonths,
+                        ValidityMonths
+                    FROM GroupCustomisations AS gc
+                    JOIN Customisations AS c ON c.CustomisationID = gc.CustomisationID
+                    INNER JOIN dbo.CentreApplications AS ca ON ca.ApplicationID = c.ApplicationID

```

Comment:
Similar to the other one where it has been set to try and line up with `GetGroupCourses`.
With reagrds to CentreApplications specifically, this join is used to join with Applications within which we do utilise and return some of that information so it seems valid enough.

Reply:
Yeah, we've had a few conversations in the team about how methods like GetGroupCourses could be improved a bit.
This method is still filtering out the groupCustomisation if it's not associated with the group or it's not useable by the centre.
I think it would be good to at least rename the method to acknowledge that. And actually, since we're getting the GroupId in the object we're returning, we could filter that in c# and simplify the sql query. So you'd end up with "GetGroupCourseIfUseableByCentre(groupCustomisationId, centreId)`.

Regarding CentreApplications - we can join directly from Customisations to Application.

Reply:
Could change the name of GetGroupCourses as well if you feel you understand it well enough.

Reply:
Done